sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -2848,6 +2861,7 @@ __ESIMD_API std::enable_if_t<
 lsc_atomic_update(T *p, __ESIMD_NS::simd<Toffset, N> offsets,
                   __ESIMD_NS::simd<T, N> src0, __ESIMD_NS::simd_mask<N> pred) {
   static_assert(std::is_integral_v<Toffset>, "Unsupported offset type");
+  static_assert(sizeof(T) > 1, "Unsupported data type");
```

Comment:
It is Ok too, but because all atomics do not support bytes, then it could be moved to check_lsc_atomic()